admin/server.go

```diff
@@ -90,6 +100,8 @@ func (s *Server) Start() {
 	rs.RegisterCodec(json.NewCodec(), "application/json;charset=UTF-8")
 
 	rs.RegisterService(s.nodeWallet, "")
+	rs.RegisterService(s.protocolUpgradeService, "upgrade")
```

Comment:
I don't think this is enough. I think you will need to implement a proper service like the one we use for `nodeWallet`.  With http request, args and reply.

Reply:
Is there a reason we aren't just going with a CLI thing like `vega propose_upgrade --release-tag=blah --block=blah` to shoot the command in like when a new node annouces itself?

Reply:
we may, none of this is set in stone. whatever ends up simpler for the validators / process manager

Reply:
done